Classes/Repository/RepositoryWebViewController.swift

```diff
@@ -0,0 +1,199 @@
+//
+//  RepositoryWebViewController.swift
+//  Freetime
+//
+//  Created by Ivan Magda on 01/03/2018.
+//  Copyright © 2018 Ryan Nystrom. All rights reserved.
+//
+
+import UIKit
+
+// MARK: RepositoryWebViewController: UIViewController
+
+final class RepositoryWebViewController: UIViewController {
+
+    // MARK: Types
+
+    private enum State {
+        case idle
+        case fetching
+        case error(String?)
+    }
+
+    // MARK: Instance Variables
+
+    private let branch: String
+    private let path: FilePath
+    private let repo: RepositoryDetails
+
+    private var state = State.idle {
+        didSet {
+            switch state {
+            case .idle:
+                activityIndicator.stopAnimating()
+                emptyView.isHidden = true
+                webView.isHidden = false
+            case .fetching:
+                activityIndicator.startAnimating()
+                emptyView.isHidden = true
+                webView.isHidden = false
+            case .error(let message):
+                activityIndicator.stopAnimating()
+                webView.isHidden = true
+                emptyView.isHidden = false
+                emptyView.label.text = message
+            }
+        }
+    }
+
+    private var resourceURL: URL? {
+        get {
+            guard let encodedPath = path.path.addingPercentEncoding(withAllowedCharacters: .urlHostAllowed) else {
+                return nil
+            }
+
+            return URL(string: "https://github.com/\(repo.owner)/\(repo.name)/raw/\(branch)/\(encodedPath)")
+        }
+    }
+
+    private let webView = UIWebView()
```

Comment:
Since our deployment target is iOS11. Shall we use `WKWebView` instead of `UIWebView`?

If you take a look at Apple doc for `WKWebView`, it says 

>Starting in iOS 8.0 and OS X 10.10, use WKWebView to add web content to your app. Do not use UIWebView or WebView.

Reply:
Thanks, I didn’t know about that.